template/src/components/pressables/BaseButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement } from 'react'
+import {
+  ActivityIndicator,
+  Insets,
+  Pressable,
+  StyleProp,
+  StyleSheet,
+  Text,
+  TextStyle,
+  View,
+  ViewStyle,
+} from 'react-native'
+import colors from '@config/ui/colors'
+import { FC } from '@utils/types'
+
+const BORDER_RADIUS = 5
+
+type RoundButtonType = 'primary' | 'secondary'
+
+export type BaseButtonPropsType = {
+  isDisabled?: boolean | null
+  onPress?: () => void
+  iconAtStart?: ReactElement
+  iconAtEnd?: ReactElement
+  text?: string | ReactElement
+  variant?: RoundButtonType
+  style?: StyleProp<ViewStyle>
+  styleText?: StyleProp<TextStyle>
+  rootStyle?: StyleProp<ViewStyle>
+  hitSlop?: null | Insets | number
+  isLoading?: boolean
```

Comment:
Let's rename this to `loading` for the sake of consistency with `disabled`

Reply:
Done